internal/rpmmd/repository.go

```diff
@@ -172,26 +171,32 @@ func NewRPMMD() RPMMD {
 	return &rpmmdImpl{}
 }
 
-func (*rpmmdImpl) FetchPackageList(repos []RepoConfig) (PackageList, error) {
+func (*rpmmdImpl) FetchPackageList(repos []RepoConfig) (PackageList, map[string]string, error) {
 	var arguments = struct {
 		Repos []RepoConfig `json:"repos"`
 	}{repos}
-	var packages PackageList
-	err := runDNF("dump", arguments, &packages)
-	sort.Slice(packages, func(i, j int) bool {
-		return packages[i].Name < packages[j].Name
+	var reply struct {
+		Checksums map[string]string `json:"checksums"`
+		Packages  PackageList       `json:"packages"`
+	}
+	err := runDNF("dump", arguments, &reply)
+	sort.Slice(reply.Packages, func(i, j int) bool {
+		return reply.Packages[i].Name < reply.Packages[j].Name
 	})
-	return packages, err
+	return reply.Packages, reply.Checksums, err
 }
 
-func (*rpmmdImpl) Depsolve(specs []string, repos []RepoConfig) ([]PackageSpec, error) {
+func (*rpmmdImpl) Depsolve(specs []string, repos []RepoConfig) ([]PackageSpec, map[string]string, error) {
 	var arguments = struct {
 		PackageSpecs []string     `json:"package-specs"`
 		Repos        []RepoConfig `json:"repos"`
 	}{specs, repos}
-	var dependencies []PackageSpec
-	err := runDNF("depsolve", arguments, &dependencies)
-	return dependencies, err
+	var reply struct {
+		Checksums    map[string]string `json:"checksums"`
+		Dependencies []PackageSpec     `json:"dependencies"`
```

Comment:
You named this field "packages" in `dnf-json`, this won't work.

Reply:
oh wow thanks. Good catch!

Our tests didn't detect this, because that only breaks the UI. We probably want to gate on those tests in the future.

Reply:
Well, we don't have tests for /blueprint/depsolve which was broken by this change. Adding them would solve it. :)

Reply:
You're completely right of course. I've added a test :)